lib/pacemaker/pcmk_sched_colocation.c

```diff
@@ -1143,26 +1143,35 @@ pcmk__apply_coloc_to_weights(pe_resource_t *dependent,
     g_hash_table_iter_init(&iter, work);
     while (g_hash_table_iter_next(&iter, NULL, (void **)&node)) {
         if (primary->allocated_to == NULL) {
-            pe_rsc_trace(dependent, "%s: %s@%s -= %d (%s inactive)",
-                         colocation->id, dependent->id, pe__node_name(node),
-                         colocation->score, primary->id);
             node->weight = pcmk__add_scores(-colocation->score, node->weight);
+            pe_rsc_trace(dependent,
+                         "Applied %s to %s score on %s (now %s after "
+                         "subtracting %s because primary %s inactive)",
+                         colocation->id, dependent->id, pe__node_name(node),
+                         pcmk_readable_score(node->weight),
+                         pcmk_readable_score(colocation->score), primary->id);
 
         } else if (pcmk__str_eq(pe_node_attribute_raw(node, attribute), value,
                                 pcmk__str_casei)) {
             if (colocation->score < CRM_SCORE_INFINITY) {
```

Comment:
Interesting that we ignore mandatory colocations (and also apply mandatory anti-colocations) when the attribute matches. I'm guessing the idea with not applying mandatory colocations is "this node remains eligible for assignment, but we still want to consider other things that would contribute to the score -- they'll all be ignored if multiple nodes have a score of `INFINITY`."

If there are two constraints with score `INFINITY / 2`, we end up in the same position, but I guess that's just a limitation of our INFINITY math, and most constraints in practice are expected to have much lower scores.